### ISSUE TITLE
Inconsistent behaviour of `argmax` between `PyTorch` and `NumPy`

### ISSUE THREAD

**Issue by chenzhekl** [pytorch/pytorch#32343]
## 🐛 Bug

In the case of multiple occurrences of the maximum values, [numpy.argmax](https://docs.scipy.org/doc/numpy/reference/generated/numpy.argmax.html) returns the indices corresponding to the first occurrence. Though not specified in the documentation, [torch.argmax](https://pytorch.org/docs/stable/torch.html#torch.argmax) returns the last occurrence instead.

## To Reproduce

```
>>> import torch
>>> a = torch.tensor([0, 1, 1, 1])
>>> print(a.argmax())
tensor(3)
```

## Expected behavior

Align the behaviors of `PyTorch` with `NumPy`, or explicitly state the differences in the documentation to avoid confusion. 

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.4.0
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.7
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration:
 - Any other relevant information:

## Additional context

cc @ngimel

**Comment by mrshenli** [comment]
Verified that the behavior is the same for CPU and GPU in PyTorch:

```python
>>> import torch
>>> a = torch.tensor([0, 1, 1, 1])
>>> print(a.cuda().argmax())
tensor(3, device='cuda:0')
>>> print(a.argmax())
tensor(3)
```

cc @peterbell10 and @pbelevich as I saw you authored the current CPU and GPU versions of `argmax_kernel_impl` :)

**Comment by ngimel** [comment]
Removing cuda label as this issue is not specific to cuda. Adding docs label as this is a documentation issue. This is expected behavior, due to parallelization we don't make any guarantees on which element is returned, so it's not even guaranteed to be the last. See also #30708

**Comment by mruberry** [comment]
Also a dupe of https://github.com/pytorch/pytorch/issues/22853.

I plan to update the documentation and describe this behavior. It is often commented upon.
